src/atomic.rs

```diff
@@ -496,7 +496,8 @@ impl<T> Atomic<T> {
     /// Bitwise "and" with the current tag.
     ///
     /// Performs a bitwise "and" operation on the current tag and the argument `val`, and sets the
-    /// new tag to the result. Returns the previous pointer.
+    /// new tag to the result. Returns the previous pointer.  Before the operation, `val` is
+    /// truncated to be fit in the bitmask returned by [`tag_bits()`](fn.tag_bits.html)).
```

Comment:
I guess it implies truncation for `and`, but it doesn't for `or`, since that wouldn't truncate it. In any case, it's not immediately obvious, so it's better to explicitly state that this is what happens. `tag_bits` being public allows the user to check that their tag fits if they want to make sure it does.